packages/api/lib/testUtils.js

```diff
@@ -1,12 +1,108 @@
 'use strict';
 
+const cloneDeep = require('lodash.clonedeep');
 const { randomString } = require('@cumulus/common/test-utils');
+const { aws: { lambda } } = require('@cumulus/common');
+const { User } = require('../models');
+
+/**
+ * Add a user that can be authenticated against
+ *
+ * @param {Object} params - params
+ * @param {User} params.userDbClient - an instance of the Users model
+ * @returns {Promise<Object>} - an object containing a userName and a password
+ */
+async function createFakeUser({ userDbClient }) {
+  // Create the user and token for this request
+  const userName = randomString();
+  const password = randomString();
+
+  await userDbClient.create([
+    {
+      userName,
+      password,
+      expires: Date.now() + (60 * 60 * 1000) // Token expires in 1 hour
+    }
+  ]);
+
+  return { userName, password };
+}
+
+/**
+ * Call the Cumulus API by invoking the Lambda function that backs the API
+ * Gateway endpoint.
+ *
+ * Intended for use with integration tests.  Will invoke the function in AWS
+ * Lambda.  This function will handle authorization of the request.
+ *
+ * @param {Object} params - params
+ * @param {string} params.prefix - the prefix configured for the stack
+ * @param {string} params.functionName - the name of the Lambda function that
+ *   backs the API Gateway endpoint.  Does not include the stack prefix in the
+ *   name.
+ * @param {string} params.payload - the payload to send to the Lambda function.
+ *   See https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-input-format
+ * @returns {Promise<Object>} - the parsed payload of the response.  See
+ *   https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-output-format
+ */
+async function callCumulusApi({ prefix, functionName, payload: userPayload }) {
+  const payload = cloneDeep(userPayload);
+
+  const userDbClient = new User(`${prefix}-UsersTable`);
+
+  const { userName, password } = await createFakeUser({ userDbClient });
+
+  // Add authorization header to the request
+  payload.headers = payload.headers || {};
+  payload.headers.Authorization = `Bearer ${password}`;
+
+  let apiOutput;
+  try {
+    apiOutput = await lambda().invoke({
+      Payload: JSON.stringify(payload),
+      FunctionName: `${prefix}-${functionName}`,
+    }).promise();
+  }
+  finally {
+    // Delete the user created for this request
+    await userDbClient.delete({ userName });
+  }
+
+  return JSON.parse(apiOutput.Payload);
+}
+
+/**
+ * Fetch a granule from the Cumulus API
+ *
+ * @param {Object} params - params
+ * @param {string} params.prefix - the prefix configured for the stack
+ * @param {string} params.granuleId - a granule ID
+ * @returns {Promise<Object>} - the granule fetched by the API
+ */
+async function getGranule({ prefix, granuleId }) {
```

Comment:
Seems like we will have a lot of these functions and `testUtils` really isn't the place for them. I think this is integration test specific, right? So maybe a new file in the integration test package? At the very least a separate file.

Reply:
Moved to `example/spec/helpers/apiTestUtils.js`